ENH: Improve robustness of Python code running in the console

lassoan opened the pull request:
When a complete string of command is executed (either copy-pasting or running from file) then we no longer split the string to lines and execute line by line because that often leads to errors. For example, parsing of this Python code line by line would fail due to the empty line in the function definition:

```
def test1():
    print("one")

    print("two")
    print("three")

def test2():
    print("ok")
```

Also make the console remember the path of the last run Python file to make it easier to run it again.

Before this fix (failure while trying to specify `test1()` function):

![image](https://user-images.githubusercontent.com/307929/197315153-e672c7d5-fd7f-4fc7-bdb2-142eeb3c7a45.png)

After the fix (everything works as expected):

![image](https://user-images.githubusercontent.com/307929/197315196-0d88425e-4480-4d54-8eae-46f44bf6291e.png)

jamesobutler commented:
Thanks! What you have described is something I’ve dealt with over the years when copy/pasting some functions.

jcfr commented:
Thanks for working on this 🙏 

 It is a great improvement that will greatly improve productivity.

Since I am reviewing on my phone, I couldn't test. That said, I spotted few typos in the comments (`more more`, `muiltiline`)

jcfr commented:
The improvements to the  runfile function make sense, especially the saving and restoring of the current command.

If systematically displaying the entire file in the console turns out to be a problem, it could be revisited to only display  the first N lines of (or first N and last N).

lassoan commented:
Thanks for the feedback. I've fixed the typos and changed the behavior when running a file to print just a message (`Execute: file/to/script.py`) instead of the file content.

lassoan commented:
I've addressed the review comments. I'll merge this now to give a chance for everyone to try this.

jcfr commented on Libs/Widgets/ctkConsole.cpp:
```diff
@@ -1040,10 +1040,32 @@ void ctkConsolePrivate::insertFromMimeData(const QMimeData* source)
     return;
     }
+
+  // Execute multiline commands at once, for more more robust command parsing and execution.
```
```suggestion
  // Execute multiline commands at once, for more robust command parsing and execution.
```

lassoan commented:
Thanks @jcfr, I've fixed one `more more` typo but apparently there were two occurrences.
